defi/src/adaptors/db-utils/adaptor-record.ts

```diff
@@ -176,23 +177,37 @@ export const storeAdaptorRecord = async (adaptorRecord: AdaptorRecord, eventTime
         }, (currentData ?? {}) as IRecordAdaptorRecordData),
         eventTimestamp
     }
-    const obj2StoreRemoveReserveKeys = Object.entries(obj2Store).reduce((acc, [key, value]) => {
-        if (dynamoReservedKeywords.includes(key.toUpperCase())) {
-            delete acc[key];
-            return acc
-        }
-        acc[key] = value
-        return acc
-    }, {} as IRecordAdaptorRecordData)
-
+    // const obj2StoreRemoveReserveKeys = Object.entries(obj2Store).reduce((acc, [key, value]) => {
+    //     if (dynamoReservedKeywords.includes(key.toUpperCase())) {
+    //         delete acc[key];
+    //         return acc
+    //     }
+    //     acc[key] = value
+    //     return acc
+    // }, {} as IRecordAdaptorRecordData)
+    adaptorRecord.data = obj2Store 
+    const record = adaptorRecord.getCleanAdaptorRecord()
+    const { PK, SK } = adaptorRecord.keys()
 
     try {
-        console.log("Storing", obj2StoreRemoveReserveKeys, adaptorRecord.keys())
-        await dynamodb.update({
-            Key: adaptorRecord.keys(),
-            UpdateExpression: createUpdateExpressionFromObj(obj2StoreRemoveReserveKeys),
-            ExpressionAttributeValues: createExpressionAttributeValuesFromObj(obj2StoreRemoveReserveKeys)
-        }) // Upsert like
+        console.log("Storing", adaptorRecord.keys())
+        if (record) {
+            await dynamodb.put(record.toItem())
+        }
+        if (errorData) {
+            Object.entries(errorData).forEach(([chain, error]) => {
+                if (dynamoReservedKeywords.includes(chain.toUpperCase())) {
+                    errorData[replaceReservedKeyword(chain)] = error
```

Comment:
ngl im unsure if this whole thing with dynamo reserved keywords is even needed at all, for put() it should work with all keys and if the issue is that later you need to use an `update` command then you can just wrap words and that will work to avoid reserved words

Reply:
I have to do this else it would be incompatible with existing records, like base chain is already stored as `base_key`